src/ui/public/vislib/lib/handler/handler.js

```diff
@@ -203,13 +204,19 @@ define(function (require) {
      * @method destroy
      */
     Handler.prototype.destroy = function () {
-      this.charts.forEach(function (chart) {
+      this.binder.destroy();
+
+      this.renderArray.forEach(function (renderable) {
+        if (_.isFunction(renderable.destroy)) {
+          renderable.destroy();
+        }
+      });
+
+      this.charts.splice(0).forEach(function (chart) {
```

Comment:
What is the purpose of the `splice` here? It essentially is just making a shallow copy of `this.charts`, but you're not modifying the array, so that doesn't seem necessary.